torch_geometric/utils/dropout.py

```diff
@@ -79,3 +82,45 @@ def dropout_adj(
         edge_index = torch.stack([row, col], dim=0)
 
     return edge_index, edge_attr
+
+
+def dropout_node(edge_index: Tensor, edge_attr: OptTensor = None,
+                 p: float = 0.5, num_nodes: Optional[int] = None,
+                 training: bool = True) -> Tuple[Tensor, OptTensor]:
+    r"""Randomly drops nodes from the adjacency matrix
+    :obj:`(edge_index, edge_attr)` with probability :obj:`p` using samples from
+    a Bernoulli distribution.
+
+    Args:
+        edge_index (LongTensor): The edge indices.
+        edge_attr (Tensor, optional): Edge weights or multi-dimensional
+            edge features. (default: :obj:`None`)
+        p (float, optional): Dropout probability. (default: :obj:`0.5`)
+        num_nodes (int, optional): The number of nodes, *i.e.*
+            :obj:`max_val + 1` of :attr:`edge_index`. (default: :obj:`None`)
+        training (bool, optional): If set to :obj:`False`, this operation is a
+            no-op. (default: :obj:`True`)
+
+    Examples:
+
+        >>> edge_index = torch.tensor([[0, 1, 1, 2, 2, 3],
+        ...                            [1, 0, 2, 1, 3, 2]])
+        >>> edge_attr = torch.tensor([1, 2, 3, 4, 5, 6])
+        >>> dropout_node(edge_index, edge_attr)
+        (tensor([[2, 3],
+                [3, 2]]),
+        tensor([5, 6]))
+    """
+    if p < 0. or p > 1.:
+        raise ValueError(f'Dropout probability has to be between 0 and 1 '
+                         f'(got {p}')
+
+    if not training or p == 0.0:
+        return edge_index, edge_attr
+
+    num_nodes = maybe_num_nodes(edge_index, num_nodes)
+    nodes = torch.arange(num_nodes, dtype=torch.long, device=edge_index.device)
+    mask = torch.full_like(nodes, 1 - p, dtype=torch.float32)
+    mask = torch.bernoulli(mask).to(torch.bool)
+    subset = nodes[mask]
+    return subgraph(subset, edge_index, edge_attr, num_nodes=num_nodes)
```

Comment:
```suggestion
    return subgraph(mask, edge_index, edge_attr, num_nodes=num_nodes)
```
`subgraph` should already support `mask`.